pom.xml

```diff
@@ -84,7 +84,7 @@
 
         <!-- Mongo modules -->
         <mongo-java-driver.version>2.11.3</mongo-java-driver.version>
-        <bson4jackson.version>2.3.1</bson4jackson.version>
+        <bson4jackson.version>2.4.0</bson4jackson.version>
```

Comment:
@HelloCoCooo as stated in [this comment](https://github.com/restx/restx/issues/297#issuecomment-466637351) I'd rather drop the explicit `bson4jackson` dependency from `restx-jongo` module and rely on `jongo`'s transitive dependency version

That way, we will be less tied to any `jongo` upgrade inconsistency

Reply:
@fcamblor This solution can fix issue #297 with minimal impact on dependency tree structure. 
I understand what you mean.  What I am concerned about is---
removing the direct dependency may bring risks when `jongo`'s dependent structure changes in the future.

Reply:
> This solution can fix issue #297 with minimal impact on dependency tree structure.
Removing `bson4jackson`'s "direct" dependency (from `restx-jongo` module) will have the exact same implication from my POV

I guess that we put a "direct dependency" [at the time we upgraded to `jongo:1.0`](https://github.com/restx/restx/commit/5226746360c8e5664abf8d9ec1d0816726a31347) as I suppose pulled `bson4jackson:2.2.3` dependency was having some bugs and we wanted to explicitely align over "the latest available `bson4jackson` version at that time (`bson4jackson:2.3.1`)

Luckily, `jongo` was "forward compatible" with this version but, as you stated, it implied a discrepancy between `jongo` dependency and `restx-jongo` dependency.

From my POV, since we're now on a "greater than `bson4jackson:2.2.3` version", then there is no need to explicitely override `jongo` transitive dependency with our own.
Worst, as you stated, it can lead to unexpected behaviour if we forget to align overriden bson4jackson explicit dependency if we don't take care of it (and I fall in this trap when upgrading to `jongo:1.0`)

That's why I'm really more in favour of removing this dep than aligning it over `jongo` transitive dep.

Reply:
Thanks for your detailed explanation. I totally agree with your suggestion.